src/lightning/pytorch/tuner/batch_size_scaling.py

```diff
@@ -63,6 +63,9 @@ def _scale_batch_size(
         rank_zero_warn("Skipping batch size scaler since `fast_dev_run` is enabled.")
         return None
 
+    # Capture whether the module is in training or eval mode
+    is_training = trainer.training
```

Comment:
I would prefer that this captures the `trainer.lightning_module.training` state instead, which can still be used to restore later with `trainer.lightning_module.train(was_training)`